crates/bevy_asset/src/asset_server.rs

```diff
@@ -59,7 +76,29 @@ pub struct AssetServerInternal {
     task_pool: TaskPool,
 }
 
-/// Loads assets from the filesystem on background threads
+/// Loads assets from the filesystem on the background.
+///
+/// The asset server is the primary way of loading assets in bevy. It keeps track of the load state
+/// of the assets it manages and can even reload them from the filesystem [if you tell it to]!
+///
+/// The asset server is a _resource_, so in order to accesss it in a system you need a `Res`
+/// accessor, like this:
+///
+/// ```rust
+/// use bevy_asset::AssetServer;
+/// use bevy_ecs::prelude::{Commands, Res};
+
+/// fn my_system(mut commands: Commands, asset_server: Res<AssetServer>)
+/// {
+///     // Now you can do whatever you want with the asset server, such as loading an asset:
+///     // let asset_handle: Handle<Image> = asset_server.load("cool_picture.png");
+/// }
+/// ```
```

Comment:
To clarify, you can use ```rust, no_run to enable that.

Reply:
I added `no_run` but it still complains about `Image` not being found with `cargo test --doc --package bevy_asset`. And if the example code getting out of sync is a concern, not running it would not be ideal. Maybe I should just alias Image to `()` at the top for the sake of the example?

Reply:
It'll compile but not run, so getting out of date isn't too bad IMO.

Reply:
As Alice says, we'll at least be alerted to API breakages like function signatures being changed.
> Maybe I should make a simple type implement Asset just for the sake of the example?

This would be good! You don't even have to make it visible in the rendered docs if you don't want:

```
/// ```rust,no_run
/// # #[derive(Debug, TypeUuid)]
/// # struct Image;
/// (etc.)
/// ```
```

That puts `Image` in scope without having to write a real definition for it.